### PR TITLE
UX-484 Add Table.SortButton Component

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Adds new `Table.SortButton` component

### How To Test or Verify
- Run libby with `npm run start:libby`
- Visit http://localhost:9001/?path=Table__all-table-components&source=false
- Verify functionality works as expected
- Verify the component matches the mocks: https://sparkpost.invisionapp.com/console/Table-Header-Sortable-Button-ckm3xhjjy019n01v2f70wc4jk/ckmf5lg4c008n01xxghkwd3vt/play

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/components/Table/SortButton.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  return (
+    <UnsortedWrapper>
+      <ArrowUpward size={14} />
+      <ArrowDownward size={14} />
+    </UnsortedWrapper>
+  );
+}
+
+const SortButton = React.forwardRef(function SortButton(props, userRef) {
+  const { children, direction, onClick, onFocus, onBlur } = props;
```
Comment (jonambas) [inline packages/matchbox/src/components/Table/SortButton.js:85]: Wonder if we need to support more standard button props here

**Review by logansparlin — APPROVED** [review]
This looks really good, one small comment but good to merge

File: packages/matchbox/src/components/Table/SortButton.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  cursor: pointer;
+  padding: ${({ theme }) => theme.space[100]} ${({ theme }) => theme.space[200]};
+  margin-left: -${({ theme }) => theme.space[200]};
+  margin-right: -${({ theme }) => theme.space[200]};
+  margin-top: -${({ theme }) => theme.space[100]};
+  margin-bottom: -${({ theme }) => theme.space[100]};
+  border: 1px solid transparent;
+  border-radius: ${({ theme }) => theme.radii[200]};
+
+  transition: ${({ theme }) => theme.motion.duration.fast};
```
[the remark this answers is not in the document]
Comment (jonambas) [reply packages/matchbox/src/components/Table/SortButton.js:24]: Yeah I forgot we also have the `css` function we can use

[the remark this answers is not in the document]
Comment (jonambas) [reply packages/matchbox/src/components/Table/SortButton.js:24]: Oh wait, I cant do this because I need to negate some of the values

**Pull request merged**